web/client/utils/GeoStoryUtils.js

```diff
@@ -496,6 +496,38 @@ export const parseHashUrlScrollUpdate = (url, hash = '', storyId, sectionId, col
     return null;
 };
 
+export const customEntityTransform = (entity, text) => {
+    if (entity.type === 'MENTION') {
+        return "<a href=\"".concat(entity.data.url, "\" class=\"wysiwyg-mention\" data-mention data-value=\"").concat(entity.data.value, "\">").concat(text, "</a>");
+    }
+
+    if (entity.type === 'LINK') {
+        const targetOption = entity.data.targetOption || '_self';
+        const dataInteraction = entity.data.geoStorySection;
```

Comment:
We need to store more information inside the attributes for `LINK` type of intreal links so instead to have entity.data.geoStorySection we should add more options with the current structure or to have an `entity.data.attributes` property.

before to see the proposal below could it be possible to use the provided [linkCallback](https://github.com/geosolutions-it/react-draft-wysiwyg/blob/258b01b0a017f11e3bd08dcb824f42071a468c64/src/controls/Link/index.js#L62) to add more inforamation to the link object? if not we should make some changes as suggested below.

`entity.data.attributes` should be an object that contain following properties:

- `data-geostory-interaction-type`: in this case `scrollTo` but we will have different types in the future
- `data-geostory-interaction-params`: arguments separated by semicolon `param1;param2;...` in this case the section or column id

the `<a/>` tag should looks linke this:

```xml
<a  data-geostory-interaction-type="scrollTo" data-geostory-interaction-params="id" onclick="__geostory_interaction(<type, param1, param2, ..., paramN)" >text</a>
```

to use this we should change also the current implementation inside `react-draft-wysiwyg` and `html-to-draftjs`.

`react-draft-wysiwyg`

- [src/controls/Link/index.js](https://github.com/geosolutions-it/react-draft-wysiwyg/blob/258b01b0a017f11e3bd08dcb824f42071a468c64/src/controls/Link/index.js#L60-L72)

```diff
- onChange = (action, title, target, targetOption, geoStorySection) => {
+ onChange = (action, title, target, targetOption, attributes) => {
    const {
        config: { linkCallback },
    } = this.props;

    if (action === 'link') {
        const linkifyCallback = linkCallback || linkifyLink;
-       const linkified = linkifyCallback({ title, target, targetOption, geoStorySection });
-       this.addLink(linkified.title, linkified.target, linkified.targetOption, linkified.geoStorySection);
+       const linkified = linkifyCallback({ title, target, targetOption, attributes });
+       this.addLink(linkified.title, linkified.target, linkified.targetOption, linkified.attributes );
    } else {
        this.removeLink();
    }
};
```

```diff
- addLink = (linkTitle, linkTarget, linkTargetOption, geoStorySection) => {
+ addLink = (linkTitle, linkTarget, linkTargetOption, attributes) => {
    const { editorState, onChange, config: { trailingWhitespace = false } } = this.props;
    const { currentEntity } = this.state;
    let selection = editorState.getSelection();

    if (currentEntity) {
        const entityRange = getEntityRange(editorState, currentEntity);
        const isBackward = selection.getIsBackward();
        if (isBackward) {
            selection = selection.merge({
                anchorOffset: entityRange.end,
                focusOffset: entityRange.start,
            });
        } else {
            selection = selection.merge({
                anchorOffset: entityRange.start,
                focusOffset: entityRange.end,
            });
        }
    }

    const entityKey = editorState
        .getCurrentContent()
        .createEntity('LINK', 'MUTABLE', {
            url: linkTarget,
            targetOption: linkTargetOption,
-           geoStorySection
+           attributes
        })
        .getLastCreatedEntityKey();
...
```

`html-to-draftjs`

this library provide a callback to change specific node but it seems it does not work as expected from some test made by @AGMETEOR so we need to:

- clone and plublish the repo
- modify [getEntityId]( https://github.com/jpuri/html-to-draftjs/blob/d8e73afcce0b0414bc60e08f38a013968fb17003/src/library/getEntityId.js#L19-L27) to retrieve and store all attributes of a node

this is the current solution by @AGMETEOR
```diff 
const getEntityId = (node) => {
       entityConfig.url = node.getAttribute ? node.getAttribute('href') || node.href : node.href;
       entityConfig.title = node.innerHTML;
       entityConfig.targetOption = node.target;
+      const dataInteraction = node.getAttribute && node.getAttribute('data-geostory-interaction');
+      if (dataInteraction) {
+        entityConfig.geoStorySection = dataInteraction;
+      }
       entityId = Entity.__create(
         'LINK',
         'MUTABLE',

```

we could change if using `node.attributes` and `entityConfig.attributes`

Probably I missed some change to apply but the main concept is to remove specific file related to MapStore or GeoStory from draft js and make the __geostory_interaction more extensible.

@offtherailz could you take a look at this?

Reply:
For reference `https://github.com/jpuri/html-to-draftjs/issues/62`